lib/shelley/src/Cardano/Wallet/Shelley/Pools.hs

```diff
@@ -0,0 +1,42 @@
+{-# LANGUAGE LambdaCase #-}
+-- |
+-- Copyright: © 2020 IOHK
+-- License: Apache-2.0
+--
+-- Haskell-node "shelley" implementation of the @StakePoolLayer@ abstraction,
+-- i.e. some boring glue.
+module Cardano.Wallet.Shelley.Pools where
+
+import Prelude
+
+import Cardano.Pool
+    ( StakePoolLayer (..) )
+import Cardano.Wallet.Api.Server
+    ( LiftHandler (..), apiError )
+import Cardano.Wallet.Api.Types
+    ( ApiErrorCode (NotImplemented) )
+import Cardano.Wallet.Network
+    ( NetworkLayer (..) )
+import Control.Monad.Trans.Except
+    ( throwE )
+import Servant
+    ( err502 )
+
+data ListStakePoolError
+    = NotYetImplemented
+  deriving (Show, Eq)
+
+newStakePoolLayer
+    :: NetworkLayer IO t b
+    -> StakePoolLayer ListStakePoolError IO
+newStakePoolLayer _ = StakePoolLayer
+    { listStakePools = throwE NotYetImplemented
+    , knownStakePools = return []
+    }
+
+instance LiftHandler ListStakePoolError where
```

Comment:
I wonder if it can be acceptable to define `LiftHandler` in this module like this. I _think_ it's neat tbh.